src/runtime/hexagon_remote/elf.h

```diff
@@ -143,7 +159,8 @@ struct elf_t {
         // stuff, the same size again to make a writeable copy, and
         // then an extra page for the global offset table for PIC
         // code.
-        buf = (char *)mmap(NULL, size * 2 + global_offset_table_size,
+        mmap_ptr = (mmap_ptr_fn) halide_get_symbol("mmap");
```

Comment:
btw, the reason i am seeing these issues now, is that we finally installed a more recent 8998 software image. There was a long period of time where the dlopenbuf and mmap/mprotect/munmap were allowed to coexist for our ease of testing. Or in this case, obscuring of a looming issue.

An aside: on the system you observed the missing munmap, did dlopenbuf appear as a known symbol?